Training24/Test.c

```diff
@@ -1,14 +1,119 @@
 // ------------------------------------------------------------------------------------------------
 // Training ~ A training program for new joiners at Metamation, Batch - July 2024.
 // Copyright (c) Metamation India.
+// Kalaimagal V P
 // ------------------------------------------------------------------------------------------------
 // Test.c
-// Program on main branch.
+// Program on A5 branch.
+// Write a program that sorts integer arrays in ascending order using any of the sorting algorithms discussed
+// in the Sorting workshop. Also, implement binary search algorithm to search for an element in the sorted array.
+// You may take the arrays as input from the user.
 // ------------------------------------------------------------------------------------------------
 #include <stdio.h>
-#include "Header.h"
+#include <ctype.h>    // For tolower
+#include <stdlib.h>   // For strtol and realloc
+#include <conio.h>    // For getch()
+#include "header.h"
 
+// ANSI escape codes for colors
+#define RESET   "\033[0m"
+#define SKYBLUE "\033[36m"
+#define GREEN   "\033[32m"
+#define YELLOW  "\033[33m"
+#define RED     "\033[31m"
+
+/// <summary>Function to clear the console screen.</summary>
+void ClearScreen () {
+#ifdef _WIN32
+   system ("cls");           // Windows-specific command to clear the screen
+#endif
+}
+
+/// <summary>Function to display the elements of an array.</summary>
+void DisplayArray (int arr[], int size) {
+   printf ("| ");
+   for (int i = 0; i < size; i++) printf ("%d  ", arr[i]);
+   printf ("|\n");
+}
+
+/// <summary>Function to format and print test results.</summary>
+void PrintTestCase (int numTests, int inputs[], int size, int searchElement) {
+   int sortedArray[50];
+   // Copy input array for sorting
+   for (int i = 0; i < size; i++) sortedArray[i] = inputs[i];
+   InsertionSort (sortedArray, size);
+   printf (YELLOW "------------------ Test Case %d -------------------\n" RESET, numTests);
+   printf ("| Input Array:            ");
+   DisplayArray (inputs, size);
+   printf ("| Output Array:           ");
+   DisplayArray (sortedArray, size);
+   printf ("Insertion sort: %s\n", GREEN "PASS" RESET);
+   int index = BinarySearch (sortedArray, size, searchElement);
+   if (index != -1) printf ("Element %d found at index %d.\n", searchElement, index);
+   else printf ("Element %d not found.\n", searchElement);
+   printf ("--------------------------------------------------\n\n");
+}
+
+/// <summary>Tests sorting and searching algorithms with predefined cases.</summary>
+void TestSort () {
+   struct {
+      int input[50], searchElement, size;
+   } tests[] = {

```

Comment:
Priorly pass the correct index after sorting with actual index in which the test element found